File: BookEcommerce/Dto/CheckoutDto.cs
namespace BookStore.Dto;
using BookStore.Models;
using BookStore.Extensions;

public class PaymentDetailDto
{
    public Guid Id { get; set; }

    public double Amount { get; set; }
    public PaymentProvider Provider { get; set; } = PaymentProvider.COD;
    public PaymentStatus Status { get; set; } = PaymentStatus.Pending;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; } = DateTime.UtcNow;
    public string Metadata { get; set; } = null!;
}
public static class PaymentDetailDtoExtension
{
    public static PaymentDetailDto AsDto(this PaymentDetail paymentDetail)
    {
        return new PaymentDetailDto
        {
            Id = paymentDetail.Id,
            Amount = paymentDetail.Amount,
            Provider = paymentDetail.Provider,
            Status = paymentDetail.Status,
            CreatedAt = paymentDetail.CreatedAt,
            UpdatedAt = paymentDetail.UpdatedAt,
            Metadata = paymentDetail.Metadata,
        };
    }
}

File: shop/src/core/types/server-dto/checkoutDto.ts
export enum PaymentProvider {
    COD = "COD",
    // Paypal = 1,
    Momo = "MOMO",
    // ZaloPay = 3,
}
export const PAYMENT_STATUS_STR = ["Đang chờ thanh toán", "Đã thanh toán", "Thanh toán thất bại"];
export enum PaymentStatus {
    Pending = 0,
    Success = 1,
    Failed = 2,
}
export interface PaymentDetailDto {
    id: string;
    amount: number;
    provider: PaymentProvider;
    status: PaymentStatus;
    createdAt: string;
    updatedAt: string | null;
    metadata: string;
}
